utils.py

```diff
@@ -103,6 +103,104 @@ def store_boe_pdfs(base_url, minutes_url):
+                  'september': '09',
+                  'october': '10',
+                  'november': '11',
+                  'december': '12'
+                }
+
+    # this creates a dictionary containing all possible single-letter deletions of every month
+    # it's used to correct for typos in the text of a link to the minutes
+    # ex: Novembe 17, 2010 (one of two real examples as of Aug 9, 2020)
+    typo_dict = {k: list() for k in month_dict.keys()}
+    for month in typo_dict.keys():
+        for i in range(len(month)):
+            typo_dict[month].append(month[:i] + month[i+1:])
+
+    start_page = requests.get(minutes_url)  # grab the 'starting page' that provides links to the minutes for a specific year
+    start_page.raise_for_status()  # doing the \
+    start_soup = BS(start_page.text, 'lxml')  # needful
+
+    # this eliminates the need to specify the years to grab since four-digit years are used consistently
+    year_tags = start_soup.find_all('a', href=True, text=re.compile(r'^\d{4}$'))  # find the tags that link to the minutes for specific years
+    year_links = [tag.get('href') for tag in year_tags]  # extracting the links
+
+    for link in year_links:
+        print(link)
+        year_page = requests.get(link)
+        year_page.raise_for_status()
+        soup = BS(year_page.text, 'lxml')
+
+        minutes_tags = soup.find_all('a', text=date_regex)  # grab tags that link to the pdfs
+
+        for tag in minutes_tags:
+            file_name_re = date_regex.search(tag.string)
+            # example filename: 'BoE Minutes 2009-04-01.pdf'
+
+            try:  # most links will fit this pattern
+                file_name = 'BoE Minutes ' + file_name_re.group(3) + '-' + month_dict[file_name_re.group(1).lower()] + '-' + file_name_re.group(2).zfill(2) + '.pdf'
+            except KeyError as e:  # this code only triggers if there's a typo in the month string
+                month_error = file_name_re.group(1).lower()
+                print(f'Error "{e}" for minutes on: "{tag.string}", regex found: {file_name_re.groups()}. Attempting typo matching...')
+                for k, v in typo_dict.items():  # this loop searches for matches among single-letter deletions
+                    correct_month = False
+                    if month_error in v:
+                        correct_month = k
+                        break
+
+                if correct_month:  # if we found a match
+                    file_name = 'BoE Minutes ' + file_name_re.group(3) + '-' + month_dict[correct_month] + '-' + file_name_re.group(2).zfill(2) + '.pdf'
+                else:
+                    #  sorry, you'll have to update the regex if you hit this code
+                    print(f'Error: could not match month string {file_name_re.group(1)} in match {file_name_re.groups()}. ')
+
+            if os.path.exists(pdf_save_dir + file_name):  # skip the download if we've already done it
+                print(f'skipping: {file_name}')
```

Comment:
Do you mean saving a .json file with the link->filename mapping? Because that could work, but the PDFs themselves have the datestamps for the meeting in the header of every page. Writing a function to check that would be fairly simple as you'd only need to look for a datestamp in the first few hundred characters, parse it, and compare to the filename. I'm fine with attempting this as a sanity check, unless you wanted to make this part of the testing framework.